analyzer.go

```diff
@@ -296,7 +296,6 @@ func (gosec *Analyzer) Check(pkg *packages.Package) {
 		gosec.context.Pkg = pkg.Types
 		gosec.context.PkgFiles = pkg.Syntax
 		gosec.context.Imports = NewImportTracker()
-		gosec.context.Imports.TrackFile(file)
 		gosec.context.PassedValues = make(map[string]interface{})
 		ast.Walk(gosec, file)
```

Comment:
...and because `Analyzer.Check()` called _both_ `TrackFile()` and `ast.Walk()`, the incorrect import name would be added (so a non-aliased import would be added, even though the package used an alias), which broke detection.

With the new code, `CheckImports()` finds both, so calling `TrackFile` is no longer needed.

I wasn't sure if that function could be removed altogether (it's slightly easier to use than starting an Analyzer, so perhaps it's useful as a utility still).

Reply:
> I wasn't sure if that function could be removed altogether (it's slightly easier to use than starting an Analyzer, so perhaps it's useful as a utility still).

Actually, perhaps it should be the reverse; use `TrackFile` once, instead of during `Visit()` (we only need it once per file). Could perhaps be as part as `Visit()` though.